packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1212,9 +1213,28 @@ function detachFiberMutation(fiber: Fiber) {
   fiber.return = null;
 }
 
-export function detachFiberAfterEffects(fiber: Fiber): void {
+export function detachFiberAfterEffects(
+  fiber: Fiber,
+  recurseIntoSibbling: ?boolean,
+): void {
   // Null out fields to improve GC for references that may be lingering (e.g. DevTools).
   // Note that we already cleared the return pointer in detachFiberMutation().
+  if (enableStrongMemoryCleanup) {
+    if (fiber.child) {
+      detachFiberAfterEffects(fiber.child, true);
+    }
+    if (fiber.sibling && recurseIntoSibbling === true) {
+      detachFiberAfterEffects(fiber.sibling, true);
+    }
+    if (fiber.stateNode) {
+      Object.keys(fiber.stateNode).forEach(key => {
```

Comment:
Would using for-in+hasOwn be better than allocating an array here, or does it not matter?